cmake-format.py

```diff
@@ -0,0 +1,74 @@
+# Licensed to the Apache Software Foundation (ASF) under one
+# or more contributor license agreements.  See the NOTICE file
+# distributed with this work for additional information
+# regarding copyright ownership.  The ASF licenses this file
+# to you under the Apache License, Version 2.0 (the
+# "License"); you may not use this file except in compliance
+# with the License.  You may obtain a copy of the License at
+#
+#   http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing,
+# software distributed under the License is distributed on an
+# "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+# KIND, either express or implied.  See the License for the
+# specific language governing permissions and limitations
+# under the License.
+
+# cmake-format configuration file
+
+# -----------------------------
+# Options affecting formatting.
+# -----------------------------
+with section("format"):
+    # How wide to allow formatted cmake files
+    line_width = 90
```

Comment:
I'd prefer 100 if we see any point of limiting the line width for CMake files at all.

Reply:
Java and iceberg-python also seem to have 100 for limit. I personally find that with 90 chars you have to break lines too often hurting code readability. Not sure what advantages it have for PR reviewing.

Reply:
`iceberg-python` is at 130: https://github.com/apache/iceberg-python/blob/ede363bd63640f3db6c3b3d6904a5765dafd2600/pyproject.toml#L916

Reply:
Yes, and you get this kind of code that's very annoying to read:
![pyiceberg1](https://github.com/user-attachments/assets/c022198d-c832-4eea-962f-3d66c5508a54)

Reply:
Obviously it depends on what device you use for reading the code, but for me, not even on a big screen but on a regular sized laptop this whole thing fits into the screen including the navigator panel on the left.

<img width="1777" alt="Screenshot 2024-12-11 at 10 33 26" src="https://github.com/user-attachments/assets/83bbc1a4-0c71-4473-9d32-1a703ee0eb43">

Reply:
I'm not sure I get you right, this cmake-format.py is for formatting CMakeLists.txt files, not python files. 

I checked some large projects' CMakeLists.txt(impala, arrow, ceph), it seems there can be some long `message` settings in the file, other than that most of the settings seems fairly short. 

I can stand both 90 and 100 but 90 is my first choice though.

Reply:
I think on the other PR we concluded (even me :) ) to use the 90 limit. This one should have the same value.